### ISSUE TITLE
`no-unsafe-any` when using typed moment library

### ISSUE THREAD

**Issue by DylanRJohnston** [palantir/tslint#3988]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
import moment from 'moment'

const foo = moment()
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:all", "tslint-config-prettier"],
  "rules": {
    "ban-comma-operator": false,
    "ban-types": {
      "options": [
        ["Object", "Avoid using the `Object` type. Did you mean `object`?"],
        ["Boolean", "Avoid using the `Boolean` type. Did you mean `boolean`?"],
        ["Number", "Avoid using the `Number` type. Did you mean `number`?"],
        ["String", "Avoid using the `String` type. Did you mean `string`?"],
        ["Symbol", "Avoid using the `Symbol` type. Did you mean `symbol`?"]
      ]
    },
    "completed-docs": false,
    "interface-name": [true, "never-prefix"],
    "no-default-export": false,
    "no-empty-interface": false,
    "no-implicit-dependencies": [true, "dev"],
    "no-magic-numbers": [true, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
    "no-unbound-method": false,
    "no-void-expression": [true, "ignore-arrow-function-shorthand"],
    "strict-boolean-expressions": false,
    "type-literal-delimiter": false,
    "typedef": false
  }
}
```

#### Actual behavior
`ERROR: 3:13  no-unsafe-any       Unsafe use of expression of type 'any'.`

#### Expected behavior
No error

The moment library is well typed, typescript doesn't complain at all during compilation, hovering over the variable in vscode shows that type is `moment.Moment` but tslint doesn't seem convinced.

**Comment by DylanRJohnston** [comment]
Turning off `no-unused-variable` in tslint removes this issue, this is fine because the Typescript compiler already catches this problem.

**Comment by DylanRJohnston** [comment]
Seems related to #2649
